gosh/framework.sh.tpl

```diff
@@ -0,0 +1,106 @@
+#!/bin/sh
+
+set -u
+
+GOSH_FAILED=0
+
+if command -v gt >/dev/null 2>&1; then
+    readonly GT_BIN="gt"
+else
+    echo "Error: gt command not found in PATH" >&2
+    exit 1
+fi
+
+# Send a signal to the test runner
+# Args:
+#   $1 - The operation to perform (e.g., start, stop, status)
+#   $2 - The name of the test
+#   $3 - The exit code of the test (optional, default is 0)
+#   $4 - An optional message to include (default is "")
+_gosh_signal() {
+    local op="${1}"
+    local test_name="${2}"
+    local exit_code="${3:-0}"
+    local message="${4:-}"
+
+    "${GT_BIN}" ctrl \
+        --op "$op" \
```

Comment:
nit. can we do `--op="$op"` here? I find `--key=value` so much more more obvious than `--key value`.  Its not so bad here, but in many cases i see something like this and don't know if `value` is a positional argument to the command or a value to the flag.